src/c++/perf_analyzer/request_rate_manager.cc

```diff
@@ -109,22 +108,55 @@ RequestRateManager::ResetWorkers()
 void
 RequestRateManager::GenerateSchedule(const double request_rate)
 {
+  std::chrono::nanoseconds duration;
   std::function<std::chrono::nanoseconds(std::mt19937&)> distribution;
+
   if (request_distribution_ == Distribution::POISSON) {
     distribution = ScheduleDistribution<Distribution::POISSON>(request_rate);
+    // Poisson distribution needs to generate a full schedule so that the result
+    // is as random and as close to the desired rate as possible
+    duration = *gen_duration_;
   } else if (request_distribution_ == Distribution::CONSTANT) {
     distribution = ScheduleDistribution<Distribution::CONSTANT>(request_rate);
+    // Constant distribution only needs one entry per worker
```

Comment:
Can you explain this comment more? One entry of what? Why does it only need one for constant?

Reply:
Discussed offline. Action item for clearer comment perhaps

Reply:
I tweaked the comment some